src/main/java/com/google/sps/servlets/CreateRoomServlet.java

```diff
@@ -0,0 +1,104 @@
+import com.google.appengine.api.datastore.DatastoreFailureException;
+import com.google.appengine.api.datastore.QueryResultList;
+import java.util.stream.Collectors;
+
+//Tested by end to end testing the creation of the new Room. 
+//This was successful and the room Id was returned to the user
+
+//Servlet that handles the creation of a new Room
+@WebServlet("/create-room")
+public final class CreateRoomServlet extends HttpServlet {
+    private static final String INVITEES_PARAMETER = "Invitees";
+    private static final String PLAYLIST_URL_PARAMETER = "PlaylistUrl";
+    private static final String ERROR_HTML = "<h1>An error occured while creating you room.</h1>";
+
+    public void doPost(HttpServletRequest req, HttpServletResponse res) throws IOException {
+        String emails = req.getParameter(INVITEES_PARAMETER).replace(" ", "");
+        List members = Arrays.asList(emails.split(",")).stream().map(Member::createNewMember).collect(Collectors.toList());
+        //Requests playlist information from YT api and transforms playlist url to video url list
+        String playlistUrl = req.getParameter(PLAYLIST_URL_PARAMETER);
+        String playlistId = playlistUrl.substring(playlistUrl.indexOf(ServletUtil.PLAYLIST_QUERY_PARAMETER)+ServletUtil.PLAYLIST_QUERY_PARAMETER.length());
+        
+        Queue<Video> videos = new LinkedList<Video>();
+
+        Room newRoom = Room.createRoom(members, videos, new LinkedList<Message>());
+
+        // Entity roomEntity = Room.toEntity(newRoom);
+        Long newRoomId = Room.toDatastore(newRoom);
+        if(newRoomId != null) {
+           res.setContentType("text/html");
+           res.getWriter().println(createHtmlString(newRoomId));
+        } 
+        else {
+            res.setContentType("text/html");
+            res.getWriter().println(ERROR_HTML);
+        }
+    }
+    
+    /**
+      * Communicates with the Youtube Data API to get playlistItem information
+      * @param playlistId the string representing the playlistId
+      * @return an arraylist of video Urls (limit is 15)
+      */
+    public void playlistIdToVideoQueue(String playlistId, Room room) throws IOException {
+        //Connect to the YouTube Data API
+        URL url = new URL(ServletUtil.YT_DATA_API_BASE_URL+ServletUtil.DATA_API_KEY+ServletUtil.YT_DATA_API_PARAMETERS+playlistId);
+        HttpURLConnection YTDataCon = (HttpURLConnection) url.openConnection();
+        YTDataCon.setRequestMethod("GET");
+        //Read the response
+        BufferedReader in = new BufferedReader(
+        new InputStreamReader(YTDataCon.getInputStream()));
+        String inputLine;
+        StringBuffer content = new StringBuffer();
+        while ((inputLine = in.readLine()) != null) {
+            content.append(inputLine);
+        }
+        in.close();
+        //Parse json for the specific data that is necessary
+        JsonObject obj = ServletUtil.PARSER.fromJson(content.toString(), JsonObject.class);
```

Comment:
InputStreamReader responseReader = new InputStreamReader(youtubeDataConnection.getInputStream(), "UTF-8");
JsonObject response = ServletUtil.PARSER.fromJson(responseReader, JsonObject.class);

I think you'll need to do responseReader.close(); but unsure.